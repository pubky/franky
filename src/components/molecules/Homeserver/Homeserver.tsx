'use client';

import { useRouter } from 'next/navigation';

import * as Molecules from '@/molecules';
import * as Atoms from '@/atoms';

export const HomeserverHeader = () => {
  return (
    <Atoms.PageHeader>
      <Molecules.PageTitle size="large">
        Choose <span className="text-brand">homeserver.</span>
      </Molecules.PageTitle>
      <Atoms.PageSubtitle>Enter your invite code to access the Pubky homeserver. </Atoms.PageSubtitle>
    </Atoms.PageHeader>
  );
};

export const HomeserverFooter = () => {
  return (
    <Atoms.FooterLinks className="pt-6">
      By creating an account on the Pubky homeserver, you agree to the <Molecules.DialogTerms />,{' '}
      <Molecules.DialogPrivacy />, and you confirm that you are <Molecules.DialogAge />
    </Atoms.FooterLinks>
  );
};

export const HomeserverNavigation = ({
  continueButtonDisabled,
  onHandleContinueButton,
  continueText,
}: {
  continueButtonDisabled: boolean;
  onHandleContinueButton: () => void;
  continueText: string;
}) => {
  const router = useRouter();

  const onHandleBackButton = () => {
    router.push('/onboarding/backup');
  };

  return (
    <Molecules.ButtonsNavigation
      className="py-6"
      onHandleBackButton={onHandleBackButton}
      onHandleContinueButton={onHandleContinueButton}
      continueButtonDisabled={continueButtonDisabled}
      continueText={continueText}
    />
  );
};
