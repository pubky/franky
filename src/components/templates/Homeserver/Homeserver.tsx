import * as Atoms from '@/atoms';
import * as Molecules from '@/molecules';
import * as Organisms from '@/organisms';

export function Homeserver() {
  return (
    <Atoms.Container size="container" className="px-6">
      <Molecules.HomeserverHeader />
      <Organisms.InviteCodeCard />
    </Atoms.Container>
  );
}
