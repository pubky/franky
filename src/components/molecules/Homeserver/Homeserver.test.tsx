import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import { HomeserverHeader, HomeserverFooter, HomeserverNavigation } from './Homeserver';

// Mock Next.js router
const mockPush = vi.fn();
vi.mock('next/navigation', () => ({
  useRouter: () => ({
    push: mockPush,
  }),
}));

// Mock molecules
vi.mock('@/molecules', () => ({
  PageTitle: ({ children, size }: { children: React.ReactNode; size?: string }) => (
    <div data-testid="page-title" data-size={size}>
      {children}
    </div>
  ),
  DialogTerms: () => <span data-testid="dialog-terms">Terms of Service</span>,
  DialogPrivacy: () => <span data-testid="dialog-privacy">Privacy Policy</span>,
  DialogAge: () => <span data-testid="dialog-age">over 18 years old.</span>,
  ButtonsNavigation: ({
    className,
    onHandleBackButton,
    onHandleContinueButton,
    continueButtonDisabled,
    continueText,
  }: {
    className?: string;
    onHandleBackButton: () => void;
    onHandleContinueButton: () => void;
    continueButtonDisabled: boolean;
    continueText: string;
  }) => (
    <div data-testid="buttons-navigation" className={className}>
      <button data-testid="back-button" onClick={onHandleBackButton}>
        Back
      </button>
      <button data-testid="continue-button" onClick={onHandleContinueButton} disabled={continueButtonDisabled}>
        {continueText}
      </button>
    </div>
  ),
}));

// Mock atoms
vi.mock('@/atoms', () => ({
  PageHeader: ({ children }: { children: React.ReactNode }) => <div data-testid="page-header">{children}</div>,
  PageSubtitle: ({ children }: { children: React.ReactNode }) => <div data-testid="page-subtitle">{children}</div>,
  FooterLinks: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="footer-links" className={className}>
      {children}
    </div>
  ),
}));

describe('Homeserver Components - Snapshots', () => {
  describe('HomeserverHeader - Snapshots', () => {
    it('matches snapshot for default HomeserverHeader', () => {
      const { container } = render(<HomeserverHeader />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('HomeserverFooter - Snapshots', () => {
    it('matches snapshot for default HomeserverFooter', () => {
      const { container } = render(<HomeserverFooter />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('HomeserverNavigation - Snapshots', () => {
    const defaultProps = {
      continueButtonDisabled: false,
      onHandleContinueButton: vi.fn(),
      continueText: 'Continue',
    };

    it('matches snapshot for default HomeserverNavigation', () => {
      const { container } = render(<HomeserverNavigation {...defaultProps} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for HomeserverNavigation with disabled continue button', () => {
      const { container } = render(<HomeserverNavigation {...defaultProps} continueButtonDisabled={true} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for HomeserverNavigation with custom continue text', () => {
      const { container } = render(<HomeserverNavigation {...defaultProps} continueText="Join Server" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
