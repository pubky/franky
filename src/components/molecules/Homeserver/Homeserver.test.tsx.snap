// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homeserver Components - Snapshots > HomeserverFooter - Snapshots > matches snapshot for default HomeserverFooter 1`] = `
<div
  class="pt-6"
  data-testid="footer-links"
>
  By creating an account on the Pubky homeserver, you agree to the 
  <span
    data-testid="dialog-terms"
  >
    Terms of Service
  </span>
  ,
   
  <span
    data-testid="dialog-privacy"
  >
    Privacy Policy
  </span>
  , and you confirm that you are 
  <span
    data-testid="dialog-age"
  >
    over 18 years old.
  </span>
</div>
`;

exports[`Homeserver Components - Snapshots > HomeserverHeader - Snapshots > matches snapshot for default HomeserverHeader 1`] = `
<div
  data-testid="page-header"
>
  <div
    data-size="large"
    data-testid="page-title"
  >
    Choose 
    <span
      class="text-brand"
    >
      homeserver.
    </span>
  </div>
  <div
    data-testid="page-subtitle"
  >
    Enter your invite code to access the Pubky homeserver. 
  </div>
</div>
`;

exports[`Homeserver Components - Snapshots > HomeserverNavigation - Snapshots > matches snapshot for HomeserverNavigation with custom continue text 1`] = `
<div
  class="py-6"
  data-testid="buttons-navigation"
>
  <button
    data-testid="back-button"
  >
    Back
  </button>
  <button
    data-testid="continue-button"
  >
    Join Server
  </button>
</div>
`;

exports[`Homeserver Components - Snapshots > HomeserverNavigation - Snapshots > matches snapshot for HomeserverNavigation with disabled continue button 1`] = `
<div
  class="py-6"
  data-testid="buttons-navigation"
>
  <button
    data-testid="back-button"
  >
    Back
  </button>
  <button
    data-testid="continue-button"
    disabled=""
  >
    Continue
  </button>
</div>
`;

exports[`Homeserver Components - Snapshots > HomeserverNavigation - Snapshots > matches snapshot for default HomeserverNavigation 1`] = `
<div
  class="py-6"
  data-testid="buttons-navigation"
>
  <button
    data-testid="back-button"
  >
    Back
  </button>
  <button
    data-testid="continue-button"
  >
    Continue
  </button>
</div>
`;
